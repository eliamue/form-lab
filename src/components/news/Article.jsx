import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description }) => (
  <div>
    <figure>
      <figcaption>
        {title} - {author}
      </figcaption>
      <p>{description}</p>
    </figure>
  </div>
);

Article.propTypes = {
  title: PropTypes.string,
  author: PropTypes.string,
  description: PropTypes.string,
};

export default Article;
