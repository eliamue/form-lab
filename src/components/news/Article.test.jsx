import React from 'react';
import { render, screen } from '@testing-library/react';
import Article from './Article';

describe('Article', () => {
  it('displays a article', () => {
    render(
      <Article
        author="cheebong"
        // eslint-disable-next-line max-len
        title="Sharknado: The Complete 6-Movie Saga in HD Digital for $4.99 at Microsoft"
        // eslint-disable-next-line max-len
        description="Microsoft has Sharknado: The Complete Saga for $4.99. TODAY only. It is NOT Movies Anywhere, so no port. Own all 6 Sharknado films â€“ the entire saga! Take the journey from the first storm, crushi ..."
      />
    );
    // eslint-disable-next-line max-len
    const article = screen.getByText('Sharknado: The Complete 6-Movie Saga in HD Digital for $4.99 at Microsoft - cheebong');
    expect(article).toMatchSnapshot();
  });
});
