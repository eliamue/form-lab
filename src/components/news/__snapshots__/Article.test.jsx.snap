// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article displays a article 1`] = `
<figcaption>
  Sharknado: The Complete 6-Movie Saga in HD Digital for $4.99 at Microsoft
   - 
  cheebong
</figcaption>
`;
